### DISCUSSION TITLE
How to filter with related models?

### DISCUSSION THREAD

**Discussion by tux-mania** [netbox-community/pynetbox#615] [Q&A] [answered]
Hi!

I'm trying to list all circuits which are linked (on Termination A) to a site.
The problem is I don't know how to do it with pynetbox.

Within the Netbox Django ORM, if I do
```
links = Circuit.objects.filter(termination_a__site__name='MySite')
```
I get the expected results.

But when I do with pynetbox
```
links = nb.circuits.circuits.filter(termination_a__site__name='MySite')
```
I get a list of aaaaall circuits in Netbox!

What am I missing?

Thanks for your help!

**Comment by markkuleinio** [comment, marked as answer]
NetBox REST API ignores unknown request parameters, like `termination_a__site__name`, thus your call requests all circuits.

See the REST API documentation of your NetBox installation for the list of available filtering parameters in `GET /api/circuits/circuits/` call (for example: https://demo.netbox.dev/api/schema/swagger-ui/#/circuits/circuits_circuits_list). Maybe there isn't a direct corresponding option in the REST API filtering parameters, so you may need to use some additional client-side logic to implement the same functionality.

**Reply by tux-mania** [reply]
Thanks a lot for your answer.
Indeed, after diving in the documentation, I understood I was using unknown parameters, but also that some search filter are not available for some properties (like __ic for site in my case).
Thanks again!
